### ISSUE TITLE
Process terminates if 'data' listener not attached to destStream.

### ISSUE THREAD

**Issue by cdeveas** [shinnn/gulp-gh-pages#74]
Found a strange bug which occurs when the gulp-gh-pages module is 'Copying files to repository'. The destStream which is created to copy files from the master repo into the .publish directory seemed to exit prematurely for me unless I added a listener for the 'data' event. 

Simply adding the following worked for me: 
```javascript 
var destStream = vinylFs.dest(repo._repo.path)
    .on('error', reject)
    .on('data', function(data) {
        // do nothing
    })
    .on('end', function() {
        resolve(repo);
    });
```

**Comment by sintell** [comment]
@shinnn, here is my thoughts on reasons behind this issue:

You invoke `vynilFs.dest` at [122](https://github.com/shinnn/gulp-gh-pages/blame/a538d858db1acf0591bd0ecccff37a10f578f39b/index.js#L112)

And `vynilFs` internally uses `through2`, at [23](https://github.com/gulpjs/vinyl-fs/blob/bf7edd37cb0207951b75879d4e07402c51ac8770/lib/dest/index.js#L23)

And then, `through2` utilizes node native `Transform` stream, which is set with transform function at [78](https://github.com/rvagg/through2/blob/78dcc3aa608b2b0c21a28110f5b8c70b16fb07fc/through2.js#L78)

And native `Transform` stream both implements writable interface and readable interface. Also, per node docs [Readable](https://nodejs.org/api/stream.html#stream_class_stream_readable) stream is created in paused state by default. And on the way through call stack there are no `{buffer: true}` option passed through.

Hope my little investigation will help you to bring a solution to our problem. Feel free to ask me if you got questions or may need any help.

**Comment by shinnn** [comment]
https://github.com/shinnn/gulp-gh-pages/issues/72#issuecomment-149535607 (deleted since unrelated)

@jakubholynet 

> Hi, how do I find out whether this is what is happening to me? (The last output line before gulp finished that I have is "[gh-pages] Copying files to repository".)

**Comment by shinnn** [comment]
Fixed in v0.5.4.
